### DISCUSSION TITLE
Import JSON File

### DISCUSSION THREAD

**Discussion by rboada04** [SigNoz/signoz#2280] [Q&A]
We have serveral microservices running on servers that don't have network access. The microservices are wired up to send trace data to a collector which is then written to JSON files using the OTel File Exporter.
We would like to be able to import the JSON trace files into Signoz running on our local machines; however, the UI doesn't seem to support importing of JSON files. 
Can you point me in the correct direction? I'm thinking we need to send the JSON data over OTLP/HTTP.
Are there plans to support the import of JSON trace, metrics, and log files in the UI?

**Comment by ankitnayan** [comment]
@rboada04 you could add `filereceiver` at the instance where you can read those files exported by `fileexporter`. Either you can forward to SigNoz after reading from the files or you can installed SigNoz where you want to read the files.

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filereceiver

**Reply by rboada04** [reply]
Thanks for the reply. I actually came across the file receiver docs and for some reason it says it currently only supports metrics: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/filereceiver/README.md?plain=1#L14

However, [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/filereceiver/README.md?plain=1#L6) it says it supports traces and logs as well...so I will give it a shot.
